extensions/narayana-jta/deployment/src/main/java/io/quarkus/narayana/jta/deployment/NarayanaJtaProcessor.java

```diff
@@ -145,11 +156,6 @@ public void build(NarayanaJtaRecorder recorder,
             defaultProperties.remove(i);
         }
         recorder.setDefaultProperties(defaultProperties);
```

Comment:
With this, in the case of a native executable, the properties will be set at build time. This is what you want for all of them?

Reply:
Good point - @mmusgrov do know if there is any property we don't want to set at build time?

Reply:
If we only expect to apply these properties at run time. It has to revert this commit.

Reply:
I'm not sure but it seems restrictive, Narayana has many useful configuration options which were added to give users runtime control over how the TM operates.

Reply:
OK, I open https://github.com/quarkusio/quarkus/pull/40310 and it looks good to make `BeanPopulator` to init at runtime. So we don't need any change on narayana side.